### ISSUE TITLE
AWS Lambda Error: glib: Error creating thread: Resource temporarily unavailable

### ISSUE THREAD

**Issue by pigletto** [lovell/sharp#1955]
I'm running Sharp on AWS Lambda to convert source images to webp format.
My Sharp related code is as simple as:

```
image = sharp(buff).withMetadata();
image = image.toFormat('webp', {quality});
res = await image.toBuffer();

```

The problem is that sometimes this fails on the .toBuffer() call with the **glib: Error creating thread: Resource temporarily unavailable** error. By sometimes I mean eg. 1400 errors and 49000 successful executions during 1 hour period.

I've seen a similar issue with the resize #1535 but in my case setting concurrency to 1 doesn't help.

I tried:

- sharp.concurrency(1)
- sharp.cache(false)
- setting environment variable UV_THREADPOOL_SIZE=1
- retrying call to the image.toBuffer()
- recreating Sharp instance on error: sharp(buff)
- increased number of concurrent Lambda executions through AWS Support
- increased Lambda memory

still nothing. Seems that after some time (as Lambda works by freezing container until the next execution) something happens that causes it runs out of available threads.

I'm clueless currently. Any hints what more can I check. Is there any way to eg. get a number of available threads left?

**Comment by lovell** [comment]
This looks like an AWS Lambda runtime problem unrelated to either sharp or libvips.

If you're running inside a VPC and talking to other AWS services such as S3 then it's worth trying without a VPC to see if that improves things, as I recall something seeing similar when DNS queries backed up.

I notice you've cross-posted at https://forums.aws.amazon.com/thread.jspa?messageID=922634&tstart=0 (requires AWS sign-in) and would agree AWS Support is the best place to ask this kind of question.

**Comment by pigletto** [comment]
Thank you. I think I've found the possible reason, but I need to monitor/test it a bit more.

I've used serverless.com framework to deploy my Lambda to AWS. It wraps my Lambda handler function with their own handler/SDK that makes monitoring possible in the serverless.com dashboard. 
Seems that when I've switched away from the serverless handler directly to my handler, the errors are gone... Possibly serverless SDK somehow drains the number of threads.
